frontend/src/components/pages/events/EventDetails.tsx

```diff
@@ -346,6 +348,31 @@ const EventDetails: React.FC<Props> = ({ eventId }) => {
                     </Typography>
                   </>
                 )}
+                <Button
+                  variant="text"
+                  href={window.URL.createObjectURL(
+                    new File(
+                      [
+                        `BEGIN:VCALENDAR\nCALSCALE:GREGORIAN\nMETHOD:PUBLISH\nPRODID:-//Test Cal//EN\nVERSION:2.0\nBEGIN:VEVENT\nUID:\nDTSTART;VALUE=DATE:${dayjs(
+                          eventData.event.startTime
+                        )
+                          .locale(nb)
+                          .format("YYYYMMDDTHHmmss")}\nDTEND;VALUE=DATE:${dayjs(endTime)
+                          .locale(nb)
+                          .format("YYYYMMDDTHHmmss")}\nSUMMARY:${
+                          eventData.event.title
+                        }\nLOCATION:${location}\nDESCRIPTION:${eventData.event.description}\nEND:VEVENT\nEND:VCALENDAR`,
+                      ],
+                      "foo.ics",
+                      {
+                        type: "text/calendar",
+                      }
+                    )
```

Comment:
I suggest moving this into a helper in a `frontend/utils/calendars.ts` file, as calendar exports might be useful elsewhere, and helps reduce the complexity of an already complex page (>> 250 lines).